Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient.m

```diff
@@ -644,23 +555,56 @@ - (NSError *)errorFromJSON:(NSDictionary *)errorDictionary statusCode:(NSInteger
 	return [[NSError alloc] initWithDomain:kShopifyError code:statusCode userInfo:errorDictionary];
 }
 
-- (NSURLSessionDataTask *)requestForURL:(NSURL *)url method:(NSString *)method object:(id <BUYSerializable>)object completionHandler:(void (^)(NSDictionary *json, NSURLResponse *response, NSError *error))completionHandler
+#pragma mark - Convenience Requests
+
+- (NSURLSessionDataTask *)getRequestForURL:(NSURL *)url completionHandler:(void (^)(NSDictionary *json, NSURLResponse *response, NSError *error))completionHandler
 {
-	NSDictionary *json = [object jsonDictionaryForCheckout];
-	NSError *error = nil;
-	NSData *data = [NSJSONSerialization dataWithJSONObject:json options:0 error:&error];
-	NSURLSessionDataTask *task = nil;
-	if (data && !error) {
-		task = [self requestForURL:url method:method body:data completionHandler:completionHandler];
-	}
-	return task;
+	return [self requestForURL:url method:kGET object:nil completionHandler:completionHandler];
 }
 
+- (NSURLSessionDataTask *)postRequestForURL:(NSURL *)url object:(id <BUYSerializable>)object completionHandler:(void (^)(NSDictionary *json, NSURLResponse *response, NSError *error))completionHandler
+{
+	return [self requestForURL:url method:kPOST object:object completionHandler:completionHandler];
+}
+
+- (NSURLSessionDataTask *)putRequestForURL:(NSURL *)url object:(id<BUYSerializable>)object completionHandler:(void (^)(NSDictionary *json, NSURLResponse *response, NSError *error))completionHandler
+{
+	return [self requestForURL:url method:kPUT object:object completionHandler:completionHandler];
+}
+
+- (NSURLSessionDataTask *)patchRequestForURL:(NSURL *)url object:(id <BUYSerializable>)object completionHandler:(void (^)(NSDictionary *json, NSURLResponse *response, NSError *error))completionHandler
+{
+	return [self requestForURL:url method:kPATCH object:object completionHandler:completionHandler];
+}
+
+- (NSURLSessionDataTask *)deleteRequestForURL:(NSURL *)url completionHandler:(void (^)(NSDictionary *json, NSURLResponse *response, NSError *error))completionHandler
+{
+	return [self requestForURL:url method:kDELETE object:nil completionHandler:completionHandler];
+}
+
+#pragma mark - Generic Requests
+
 - (void)startTask:(NSURLSessionDataTask *)task
 {
 	[task resume];
 }
 
+- (NSString *)authorizationHeader
+{
+	NSData *data = [_apiKey dataUsingEncoding:NSUTF8StringEncoding];
+	return [NSString stringWithFormat:@"%@ %@", @"Basic", [data base64EncodedStringWithOptions:0]];
+}
+
+- (NSURLSessionDataTask *)requestForURL:(NSURL *)url method:(NSString *)method object:(id <BUYSerializable>)object completionHandler:(void (^)(NSDictionary *json, NSURLResponse *response, NSError *error))completionHandler
+{
+	NSData *data = nil;
+	if (object) {
+		data = [NSJSONSerialization dataWithJSONObject:[object jsonDictionaryForCheckout] options:0 error:nil];
+	}
+	
+	return [self requestForURL:url method:method body:data completionHandler:completionHandler];
```

Comment:
This is now the only method that calls the "body/data" version. Should we merge them?